builtin/logical/pki/path_config_ca.go

```diff
@@ -48,24 +470,57 @@ func (b *backend) pathCAWrite(
 		parsedBundle.CertificateBytes = parsedBundle.IssuingCABytes
 	}
 
-	// TODO?: CRLs can only be generated with RSA keys right now, in the
-	// Go standard library. The plubming is here to support non-RSA keys
-	// if the library gets support
+	cb := &certutil.CertBundle{}
+	entry, err := req.Storage.Get("config/ca_bundle")
+	if err != nil {
+		return nil, err
+	}
+	if entry != nil {
+		err = entry.DecodeJSON(cb)
+		if err != nil {
+			return nil, err
+		}
+		// If we have a stored private key and did not get one, attempt to
+		// correlate the two -- this could be due to a CSR being signed
+		// for a generated CA cert and the resulting cert now being uploaded
+		if len(cb.PrivateKey) != 0 &&
```

Comment:
As with all PKI things, it's rather verbose, but it's pretty straightforward: if they submit a certificate and no private key, this simply checks whether we already have a private key matching the one used in the subject of the certificate. This means that the certificate was issued to the private key we already have in storage.

If there is a mismatch between the certificate's key and the stored key, no assignment happens, and in the next statement we error out.